river/command/move.zig

```diff
@@ -0,0 +1,196 @@
+            @intCast(i32, output_box.width - view.pending.box.width) - border_width,
+    }
+
+    apply(view);
+}
+
+pub fn resize(
+    allocator: *std.mem.Allocator,
+    seat: *Seat,
+    args: []const []const u8,
+    out: *?[]const u8,
+) Error!void {
+    if (args.len < 3) return Error.NotEnoughArguments;
+    if (args.len > 3) return Error.TooManyArguments;
+
+    const delta = try std.fmt.parseInt(i32, args[2], 10);
+    const orientation = std.meta.stringToEnum(Orientation, args[1]) orelse
+        return Error.InvalidOrientation;
+
+    const view = get_view(seat) orelse return;
+    const output_box = get_output_dimensions(view);
+    const border_width = @intCast(i32, view.output.root.server.config.border_width);
+    switch (orientation) {
+        .horizontal => {
+            var real_delta: i32 = @intCast(i32, view.pending.box.width);
+            if (delta > 0) {
+                view.pending.box.width += @intCast(u32, delta);
+            } else {
+                // Prevent underflow
+                view.pending.box.width -=
+                    std.math.min(view.pending.box.width, @intCast(u32, -1 * delta));
+            }
+            view.applyConstraints();
+            // Do not grow bigger than the output
+            view.pending.box.width = std.math.min(
+                view.pending.box.width,
+                output_box.width - @intCast(u32, 2 * border_width),
+            );
+            real_delta -= @intCast(i32, view.pending.box.width);
+            view.pending.box.x = std.math.clamp(
+                view.pending.box.x + @divFloor(real_delta, 2),
+                border_width,
+                @intCast(i32, output_box.width - view.pending.box.width) - border_width,
+            );
+        },
+        .vertical => {
+            var real_delta: i32 = @intCast(i32, view.pending.box.height);
+            if (delta > 0) {
+                view.pending.box.height += @intCast(u32, delta);
+            } else {
+                // Prevent underflow
+                view.pending.box.height -=
+                    std.math.min(view.pending.box.height, @intCast(u32, -1 * delta));
+            }
+            view.applyConstraints();
+            // Do not grow bigger than the output
+            view.pending.box.height = std.math.min(
+                view.pending.box.height,
+                output_box.height - @intCast(u32, 2 * border_width),
+            );
+            real_delta -= @intCast(i32, view.pending.box.height);
+            view.pending.box.y = std.math.clamp(
+                view.pending.box.y + @divFloor(real_delta, 2),
+                border_width,
+                @intCast(i32, output_box.height - view.pending.box.height) - border_width,
+            );
+        },
+    }
+
+    apply(view);
+}
+
+fn apply(view: *View) void {
+    // Set the view to floating but keep the position and dimensions
+    view.pending.float = true;
+    view.float_box = view.pending.box;
```

Comment:
This should be set when switching away from not to floating.
```suggestion
```

Reply:
That is needed because otherwise the view will not keep the dimensions.